File: transpose.go
package main
import (
	"fmt"
)
func transpose(arr [][]int) [][]int{
	var COL_LENGTH  int = len(arr[0])
	var ROW_LENGTH	int = len(arr)

	transpose_arr := [][]int{}

	for col:=0; col < COL_LENGTH; col++{
		temp_arr := [] int{}
		for row := 0; row < ROW_LENGTH; row++{
			temp_arr = append(temp_arr, arr[row][col])
		}
		transpose_arr = append(transpose_arr, temp_arr)
	}

	return transpose_arr

}

func main() {
	arr := [][]int{  
		{0, 1, 2, 3},  
		{4, 5, 6, 7},  
		{8, 9, 10,11},
		{10,11,12,13},
	}

	transpose_arr := transpose(arr)
	fmt.Println(transpose_arr)
}

File: Transpose.java

public class Transpose {
    public int [][] transpose(int [][] arr){
        int colLength = arr[0].length;
        int rowLength = arr.length;

        int [][] transposeArr = new int[colLength][rowLength];
        for(int col = 0; col < colLength; col++){
            for(int row = 0; row < rowLength; row++){
                transposeArr[col][row] = arr[row][col];
            }
        }
        return transposeArr;
    }

    public static void main(String[] args){
        int[][] arr = {
                { 1, 2, 10, 100},
                { 3, 4, 11, 101},
                { 5, 6, 12, 102},
                { 7, 8, 13, 103},
                { 9, 10,14, 104},
        };

        Transpose transpose = new Transpose();
        int arrResult [][] = transpose.transpose(arr);

        for (int row = 0; row < arrResult.length; row++){
            for ( int col = 0; col < arrResult[row].length; col++){
                System.out.printf("%3d ",arrResult[row][col]);
            }
            System.out.println();
        }
    }
}
